src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -157,6 +169,185 @@ namespace OpenMS
+    ConvexHull2D::PointArrayType & hull_points,
+    double & intensity_sum, 
+    double & intensity_integral,
+    double & rt_sum,
+    double & peak_apex_int,
+    double & peak_apex_rt)
+  {
+    intensity_sum = 0.0;
+    rt_sum = 0.0;
+    peak_apex_int = -1;
+    double peak_apex_dist = std::fabs(chromatogram.begin()->getMZ() - peak_apex_rt);
+    int peak_nr (0);
+    bool added_right(false);
+    // FEATURE : use RTBegin / MZBegin -> for this we need to know whether the template param is a real chromatogram or a spectrum!
+    MSChromatogram::const_iterator prev_it = chromatogram.begin();
+    for (MSChromatogram::const_iterator it = chromatogram.begin(); it != chromatogram.end(); ++it)
+    {
+      if (it->getMZ() > best_left && it->getMZ() < best_right)
+      {
+        if (peak_nr == 0 && it != chromatogram.begin())
+        {
+          // add area between first measured peak inside the peak boundaries and start of peak to the left
+          double delta_rt = it->getRT() - best_left;
+          double interpol_intensity = lin_interpolate(best_left, prev_it->getRT(), it->getRT(), prev_it->getIntensity(), it->getIntensity());
+          intensity_integral += (interpol_intensity + it->getIntensity())/2.0 * delta_rt;
+        }
+
+        if (peak_nr > 0)
+        {
+          // add area between last peak (inside boundaries) and current peak (inside boundaries)
+          double delta_rt = it->getRT() - prev_it->getRT();
+          intensity_integral += (prev_it->getIntensity() + it->getIntensity())/2.0 * delta_rt;
+        }
+
+        DPosition<2> p;
+        p[0] = it->getMZ();
+        p[1] = it->getIntensity();
+        hull_points.push_back(p);
+        if (std::fabs(it->getMZ() - peak_apex_rt) <= peak_apex_dist)
+        {
+          peak_apex_int = p[1];
+          peak_apex_dist = std::fabs(it->getMZ() - peak_apex_rt);
+        }
+        rt_sum += it->getMZ();
+        intensity_sum += it->getIntensity();
+        peak_nr++;
+      }
+      else if (peak_nr > 0 && !added_right)
+      {
+        // add area between last measured peak inside the peak boundaries and end of peak to the right
+        double delta_rt = best_right - prev_it->getRT();
+        double interpol_intensity = lin_interpolate(best_right, prev_it->getRT(), it->getRT(), prev_it->getIntensity(), it->getIntensity());
+        intensity_integral += (prev_it->getIntensity() + interpol_intensity)/2.0 * delta_rt;
+        added_right = true;
+      }
+      prev_it = it;
+    }
+  }
+
+  void MRMTransitionGroupPicker::calculatePeakShapeMetrics_(const MSChromatogram& chromatogram, 
+    double best_left, double best_right, 
+    double peak_height, double peak_apex_rt, double avg_noise_level,
+    PeakShapeMetrics_ & peakShapeMetrics)
+  {
+    peakShapeMetrics.points_across_baseline = 0;
+    double start_intensity(0), end_intensity(0);
+    double delta_rt, delta_int, height_5, height_10, height_50;
+    
+    // pass 1
```

Comment:
does this comment have any significance? is there a pass2 ?

Reply:
Nope.  That was there from a previous version I had.